File: app/src/main/java/com/example/fithealth/Model/BaseDeDatos/FitHealthDatabase.java
package com.example.fithealth.Model.BaseDeDatos;


import android.content.Context;

import androidx.annotation.NonNull;
import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;
import androidx.room.migration.Migration;
import androidx.sqlite.db.SupportSQLiteDatabase;

import com.example.fithealth.Model.BaseDeDatos.Dao.DaoUsuario;
import com.example.fithealth.Model.BaseDeDatos.Tables.Usuario.DatosUsuario;




@Database(entities = {DatosUsuario.class},version = 2,exportSchema = false)
public abstract class FitHealthDatabase extends RoomDatabase {

    private static volatile FitHealthDatabase INSTANCE;

    public abstract DaoUsuario daoUsuario();

    public static FitHealthDatabase getInstance(Context context) {

        if (INSTANCE == null) {
            synchronized (FitHealthDatabase.class) {
                if (INSTANCE == null) {
                    INSTANCE = Room.databaseBuilder(context.getApplicationContext(),FitHealthDatabase.class,"FitHealthBase").build();
                }
            }
        }
        return INSTANCE;
    }

    private static final Migration MIGRATION_1_2 = new Migration(1, 2) {
        @Override
        public void migrate(@NonNull SupportSQLiteDatabase database) {
            // Elimina la columna 'id' de la tabla 'usuario'
            database.execSQL("ALTER TABLE usuario ADD COLUMN id");

        }
    };

}

File: app/src/main/java/com/example/fithealth/Model/Database/FitHealthDatabase.kt
package com.example.fithealth.Model.Database


import androidx.room.Database
import androidx.room.RoomDatabase
import androidx.room.TypeConverters
import com.example.fithealth.Model.Database.Dao.ContactsDao
import com.example.fithealth.Model.Database.Dao.MealDao
import com.example.fithealth.Model.Database.Dao.MessageDao
import com.example.fithealth.Model.Database.Dao.UserDao
import com.example.fithealth.Model.Database.Tables.Entitys.Contact
import com.example.fithealth.Model.Database.Tables.Entitys.Meal
import com.example.fithealth.Model.Database.Tables.Entitys.Message
import com.example.fithealth.Model.Database.Tables.Entitys.User

@Database(
    entities = [Contact::class, User::class, Message::class, Meal::class],
    version = 4
)
@TypeConverters(ConvertersDatabase::class)
abstract class FitHealthDatabase : RoomDatabase() {
    abstract fun getContactsDao(): ContactsDao


    abstract fun getUserDao(): UserDao

    abstract fun getMessageDao(): MessageDao

    abstract fun getMealDao(): MealDao


}